pkg/apis/hive/v1alpha1/hiveconfig_types.go

```diff
@@ -31,6 +31,9 @@ type HiveConfigSpec struct {
 	// a cluster deployment(if specified), with precedence given to the contents of the pull secret for the cluster deployment.
 	// +optional
 	GlobalPullSecret *corev1.LocalObjectReference `json:"globalPullSecret,omitempty"`
+
+	// FailedProvisionConfig is used to configure settings related to handling provision failures.
+	FailedProvisionConfig *FailedProvisionConfig `json:"failedProvisionConfig,omitempty"`
```

Comment:
I recommend that we not make this field a pointer and `omitempty`. As it is, the default behavior is to *not* gether logs. I think we want the default behavior to be that logs will be gathered. Along those lines, I would like to see the `failedProvisionConfig.gatherLogs` field changed to `failedProvisionConfig.noLogsGather`.